gaphor/SysML/blocks/proxyport.py

```diff
@@ -35,6 +40,15 @@ def update_shapes(self):
                     self.subject, [self.diagram.gettext("proxy")]
                 )
             ),
-            Text(text=lambda: self.subject and self.subject.name or ""),
+            Text(text=self._format_name),
             style=text_position(self.connected_side()),
         )
+
+    def _format_name(self):
+        if not self.subject:
+            return ""
+
+        name = self.subject.name or ""
+        if self.show_type and self.subject.type:
+            return f"{name}: {self.subject.type.name or ''}"
```

Comment:
I think here and in the Model Navigation, it should be `f"{name} : {self.subject.type.name}`, with a space before the colon.

![image](https://github.com/gaphor/gaphor/assets/10014976/91d85e52-77a5-4d7b-bf36-df323e04bd6b)

Reply:
Really? We do not do it for attributes and operations in classes.

Reply:
I think we are mixing the model element and the presentation of the model element.

- The model element shall have a name because it is part of the model and we want to know what it is
- Whether we want to display it on a view (diagram) and in what way (only name, only type, both) is a totally different question

So, I would say that the user shall be able to configure those things **for the presentation element**.

Reply:
> Really? We do not do it for attributes and operations in classes.

I think that is how it is specified in both UML / SysML for all types including in the presentation.

> I think we are mixing the model element and the presentation of the model element.

There is no requirement in UML and SysML that things should have names. I don't understand the use case for why you would want to name something but then hide the name. I realize that MagicDraw allows that, but we aren't building a MagicDraw clone.

Reply:
My reasoning is to enable the user to be able to use a diagram really as a view. On a view I would like to display a representation of a model element a way it supports the communication I would like to achieve. Sometimes it makes sense to hide name or type, simply because it helps the communication by removing details that are deemed not important at that point when a diagram is being presented to the users. I find it totally valuable and use it day to day. It is not just magicdraw, you can have it in each and every modeling tool (EA, Visual Paradigm, and so on).

Reply:
@sz332 We understand that diagrams are views. The point is adding more complexity to configuration makes the tool harder to use, sometimes that's a good tradeoff, in this case I don't understand why you would want to show types and hide names. If you don't care what things are named on the diagrams, then why would you care that they have a name at all? I don't get it. It is also consistent, if something has a name, you know that it is automatically displayed all the time. With those other tools, you see something only has a type displayed, and you don't know if that is because the Element has no name in the model or because the name is hidden.

Reply:
But the question is: do you really want to automatically display names all the time? For example in case of pins, based on the abstraction level I am working on, I sometimes only display name, then both name and type, and in the final step I hide names where it is not important in order to decrease the noise on the diagram. So it might be a way of doing the modeling process, maybe other people are not doing it that way. 

I would say that we will have this problem in general for multiple type of elements, and I suggest to create a common solution and understanding for it (and probably adding some common code that can be easily used across all those components might make sense).

Reply:
Let's stick with names. If that has to change, it has to change throughout the app, which is a fair bit of work.

![image](https://github.com/gaphor/gaphor/assets/10014976/91d85e52-77a5-4d7b-bf36-df323e04bd6b)

In this image there's no space between `p1` and the colon, IMHO.

Reply:
I can open a separate issue about the colons since it is outside this PR. I agree that the text in the PDF is hard to tell.